lib/Dialect/Substrait/IR/Substrait.cpp

```diff
@@ -81,6 +81,21 @@ void printCountAsAll(OpAsmPrinter &printer, Operation *op, IntegerAttr count) {
   // Normal integer.
   printer << count.getValue();
 }
+
+//===----------------------------------------------------------------------===//
+// Substrait attributes
+//===----------------------------------------------------------------------===//
+
+/// Implement the getType method for custom type `TimestampAttr`.
+::mlir::Type TimestampAttr::getType() const {
```

Comment:
You *could* also use an [`extraClassDefinition`](https://mlir.llvm.org/docs/DefiningDialects/Operations/#extra-definitions) and define this one-line function inline in the TD file.